fs/config/config.go

```diff
@@ -423,14 +424,36 @@ func SaveConfig() {
 	}
 
 	if err = os.Rename(ConfigPath, ConfigPath+".old"); err != nil && !os.IsNotExist(err) {
-		log.Fatalf("Failed to move previous config to backup location: %v", err)
+		return errors.Errorf("Failed to move previous config to backup location: %v", err)
 	}
 	if err = os.Rename(f.Name(), ConfigPath); err != nil {
-		log.Fatalf("Failed to move newly written config from %s to final location: %v", f.Name(), err)
+		return errors.Errorf("Failed to move newly written config from %s to final location: %v", f.Name(), err)
 	}
 	if err := os.Remove(ConfigPath + ".old"); err != nil && !os.IsNotExist(err) {
 		fs.Errorf(nil, "Failed to remove backup config file: %v", err)
 	}
+	return nil
+}
+
+func SaveConfig() {
+	err := saveConfig()
+	if err == nil {
+		return
+	}
+
+	startTime := time.Now()
+	for i := 0; i < fs.Config.LowLevelRetries; i++ {
+		waitingTime := mathrand.Intn( int(time.Since(startTime).Seconds()))
```

Comment:
This looks over complicated to me.  I think this should do it?

```go
waitingTimeMs := mathrand.Intn(1000)
time.Sleep(time.Millisecond*waitTimeMs)
```

I'd put the sleep after the call to SaveConfig() then you can get rid of calling `saveConfig` outside of the loop.